torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -86,6 +87,23 @@ static int64_t floordiv(int64_t a, int64_t b) {
   }
 }
 
+// reference function THPVariable_to in python_variable_methods.cpp
+static at::Tensor to_dispatch(at::Tensor self, c10::optional<at::Device> device,
+      c10::optional<at::ScalarType> scalarType, bool non_blocking, bool copy) {
+  if (device && device->is_cuda()) {
+    at::globalContext().lazyInitCUDA();
+  }
+  if (!device && !scalarType && !copy) {
+    return self;
+  } else if (!device) {
```

Comment:
Is it possible for both `device` and `scalarType` to be `nullopt`s?

Reply:
I was just replicating the exiting logic which is: 
"""  
if (!device && !scalarType && !copy) {
    Py_INCREF(self);
    return self;
  } else if (!device) {
    return THPVariable_Wrap(dispatch_to(self_, *scalarType, non_blocking, copy));
  } else if (!scalarType) {
    return THPVariable_Wrap(dispatch_to(self_, *device, non_blocking, copy));
  } else {
    return THPVariable_Wrap(dispatch_to(self_, *device, *scalarType, non_blocking, copy));
  }
""" 

I could explicitly check also check for the case where copy=False and device & scalarType aren't set. That wouldn't really make sense in  user code

Reply:
It errors in pytorch, so not much point in supporting it in script:
```
>>> a.to(copy=True)
Assertion failed: (initialized()), function operator*, file ../c10/util/Optional.h, line 596.
```